src/cloudWatchLogs/timeFilterSubmenu.ts

```diff
@@ -0,0 +1,171 @@
+/*!
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { Prompter, PromptResult } from '../shared/ui/prompter'
+import { InputBoxPrompter } from '../shared/ui/inputPrompter'
+import { ItemLoadTypes, QuickPickPrompter, DataQuickPickItem, createQuickPick } from '../shared/ui/pickerPrompter'
+import { createInputBox } from '../shared/ui/inputPrompter'
+import { isValidResponse, StepEstimator } from '../shared/wizards/wizard'
+
+export function isViewAllEvents(response: TimeFilterResponse) {
+    return response.start === response.end
+}
+
+export interface TimeFilterResponse {
+    // # of miliseconds since january 1, 1970 since thats what API expects.
+    readonly start: number
+    readonly end: number
+}
+
+const customRange = Symbol('customRange')
+
+export class TimeFilterSubmenu extends Prompter<TimeFilterResponse> {
+    // TODO: Generalize submenu code between this and the region Submenu.
+    private currentState: 'custom-range' | 'recent-range' = 'recent-range'
+    private steps?: [current: number, total: number]
+    public defaultPrompter: QuickPickPrompter<typeof customRange | number> = this.createMenuPrompter()
+    public customPrompter: InputBoxPrompter = this.createDateBox()
+
+    public constructor() {
+        super()
+    }
+
+    private get recentTimeOptions(): ItemLoadTypes<number> {
+        const options: DataQuickPickItem<number>[] = []
+        options.push({
+            label: 'View all events',
```

Comment:
The name of this item defeats [parallel form](http://guidetogrammar.org/grammar/parallelism.htm), which makes the list less intuitive to scan for the user. We could preserve parallel form by naming this "Forever" or something like that, then it matches the form of the other relative-time items.